src/evict/evict_page.c

```diff
@@ -44,6 +44,36 @@ __evict_exclusive(WT_SESSION_IMPL *session, WT_REF *ref)
     return (__wt_set_return(session, EBUSY));
 }
 
+/*
+ * __wt_page_evict_soon_check --
+ *     Check whether the page should be evicted urgently.
+ */
+bool
+__wt_page_evict_soon_check(WT_SESSION_IMPL *session, WT_REF *ref, bool *inmem_split)
+{
+    WT_BTREE *btree;
+    WT_PAGE *page;
+
+    btree = S2BT(session);
+    page = ref->page;
+
+    /*
+     * Attempt to evict pages with the special "oldest" read generation. This is set for pages that
+     * grow larger than the configured memory_page_max setting, when we see many deleted items, and
+     * when we are attempting to scan without trashing the cache.
+     *
+     * Checkpoint should not queue pages for urgent eviction if they require dirty eviction: there
+     * is a special exemption that allows checkpoint to evict dirty pages in a tree that is being
+     * checkpointed, and no other thread can help with that. Checkpoints don't rely on this code for
+     * dirty eviction: that is handled explicitly in __wt_sync_file.
+     */
+    if (WT_READGEN_EVICT_SOON(page->read_gen) && btree->evict_disabled == 0 &&
```

Comment:
I remember we discussed to set the read_gen to evict soon if the page is forced evicted. I don't see that change. Is the current change enough to get back the performance?

Reply:
Updated in [38c143e](https://github.com/wiredtiger/wiredtiger/pull/7687/commits/38c143eaf7eeda67175d6b03c160741fdbf4aae7)

Reply:
I also ran the [snapshot reads](https://spruce.mongodb.com/task/sys_perf_linux_3_node_replSet_snapshot_reads_patch_0c289c9f9ea159650a9598eaeffa178c2a310544_62539befe3c33163f22303c8_22_04_11_03_09_43/trend-charts?execution=0&sortBy=STATUS&sortDir=ASC) test, and I can confirm that we got the performance back.